### ISSUE TITLE
aws_autoscaling_group termination_policies order is applied in reverse?

### ISSUE THREAD

**Issue by erkolson** [hashicorp/terraform#4532]
When I put something like this in a .tf file:
```
resource aws_autoscaling_group "some-asg" {
  name = some-asg
  vpc_zone_identifier = ["${aws_subnet.private.0.id}"]
  max_size = 2
  min_size = 2
  health_check_grace_period = 300
  health_check_type = "ELB"
  termination_policies = ["OldestLaunchConfiguration", "ClosestToNextInstanceHour"]
...
```

The order the policies are displayed in the console and apparently applied is "ClosestToNextInstanceHour", then "OldestLaunchConfiguration"

**Comment by jen20** [comment]
Hi @erkolson! Thanks for opening this issue. It's a bit unclear as to the exact context here: could you post a bit more of the configuration context and the results you'd expect and we can try to help. Thanks!

**Comment by erkolson** [comment]
Hello @jen20, I added some more configuration in the code snippet.  Given that ```termination_policies``` is an array, I was expecting the policies to be applied in order that they are arranged in the array.  The order in which the policies are applied can change which instances get terminated, so is pretty important.

**Comment by jen20** [comment]
Thanks for the update @erkolson - I'll investigate this and get back to you.

**Comment by pforman** [comment]
@erkolson just as a question, what version(s) of Terraform is this happening in?  It's not clear from the resource example.

I wrote the last PR for that (https://github.com/hashicorp/terraform/pull/2890), and it definitely _should_ be ordering them correctly.  That was the reason I wrote it, the previous behavior was "hash-based ordering", which was unpredictable and not configurable for specific ordering.

That PR was merged and eventually released in 0.6.7, so if there's an issue in 0.6.7 or newer, I can dig back in to fix it and see why the tests didn't catch it.

**Comment by erkolson** [comment]
it was indeed 0.6.7.  Thanks!

**Comment by pforman** [comment]
So does it work for you on 0.6.7 or newer?  (if you have a chance to upgrade...)

I used your test case policies on a new ASG I was building today, and saw expected behavior (on 0.6.10), including permuting them, and also passing a CSV string from tfvars into split(), which is my usual method of defining them.

**Comment by erkolson** [comment]
Using 0.6.10 now, I see the order is applied in AWS in the order specified in array the .tf file.  Thanks!

**Comment by stack72** [comment]
Hi all

This has been fixed since we ordered lists in 0.6.10 :)

Paul
